src/features/AppletSettings/ShareAppletSetting/ShareAppletSetting.tsx

```diff
@@ -14,13 +14,19 @@ import { StyledHeadline } from '../AppletSettings.styles';
 export const ShareAppletSetting = ({ isDisabled: isDisabledSetting = false }) => {
   const { t } = useTranslation('app');
   const { id } = useParams();
-  const applet = folders.useApplet(id as string);
 
   const [isSubmitted, setIsSubmitted] = useState(false);
   const [isDisabled, setIsDisabled] = useState(true);
   const [sharePopupVisible, setSharePopupVisible] = useState(false);
   const [keywords, setKeywords] = useState<string[]>([]);
   const [libraryUrl, setLibraryUrl] = useState('');
+  const [applet, setApplet] = useState<FolderApplet | null>(null);
+
+  useEffect(() => {
+    if (id) {
+      setApplet(folders.useApplet(id));
+    }
+  });
```

Comment:
maybe id needs to be added to array of dependencies

Reply:
and where the dep array?)